bundles/org.openhab.binding.mqtt.generic/src/main/java/org/openhab/binding/mqtt/generic/values/ColorValue.java

```diff
@@ -116,17 +137,31 @@ public String getMQTTpublishValue(@Nullable String pattern) {
 
         String formatPattern = pattern;
         if (formatPattern == null || "%s".equals(formatPattern)) {
-            formatPattern = "%1$d,%2$d,%3$d";
+            if (this.colorMode.equals(COLOR_MODE.xyY)) {
+                formatPattern = "%1$f,%2$f,%3$.2f";
+            } else {
+                formatPattern = "%1$d,%2$d,%3$d";
+            }
         }
-        // ignore the pattern. Don't know
-        if (isRGB) {
-            PercentType[] rgb = ((HSBType) state).toRGB();
-            return String.format(formatPattern, rgb[0].toBigDecimal().multiply(factor).intValue(),
-                    rgb[1].toBigDecimal().multiply(factor).intValue(),
-                    rgb[2].toBigDecimal().multiply(factor).intValue());
+
+        HSBType hsb_state = (HSBType) state;
+
+        switch (this.colorMode) {
+            case hsb:
+                return String.format(formatPattern, hsb_state.getHue().intValue(), hsb_state.getSaturation().intValue(),
+                        hsb_state.getBrightness().intValue());
+            case rgb:
+                PercentType[] rgb = hsb_state.toRGB();
+                return String.format(formatPattern, rgb[0].toBigDecimal().multiply(factor).intValue(),
+                        rgb[1].toBigDecimal().multiply(factor).intValue(),
+                        rgb[2].toBigDecimal().multiply(factor).intValue());
+            case xyY:
+                PercentType[] xyY = hsb_state.toXY();
+                return String.format(Locale.ROOT, formatPattern, xyY[0].floatValue() / 100.0f,
+                        xyY[1].floatValue() / 100.0f, hsb_state.getBrightness().floatValue());
+            default:
+                logger.warn("Non supported color mode: {}", this.colorMode);
+                return "";
```

Comment:
We should throw an exception (fail fast).
Returning an empty string will defere the error to a later stage and makes it harder to find the problem

Reply:
Hi @jochen314!

An empty string is also returned by the ```getMQTTpublishValue``` method when the value state is ```UnDefType.UNDEF```, but I will throw the exception anyway :wink: